### PR TITLE
feat(stdlib/universe): add elapsed to flux

### LINKED ISSUE #1429: elapsed() transformation for Flux from InfluxQL

### PR DESCRIPTION
**Pull request by ewendai**
### Done checklist
- [x] docs/SPEC.md updated
- [x] Test cases written

Close #1429

### CODE REVIEW

File: docs/SPEC.md [hunk 1]
```diff
@@ -3019,6 +3019,25 @@ from(bucket: "telegraf/autogen")
     |> filter(fn: (r) => r._measurement == "cpu" and r._field == "usage_user")
     |> difference()
 ```
+#### Elapsed (Proposal)
+
+Elapsed returns the elapsed time between subsequent records. 
+
+Given an input table, `elapsed` will return the same table with an additional `elapsed` column and without the first
```
[the remark this answers is not in the document]
Comment (nathanielc) [reply docs/SPEC.md]: What do derivative and difference do? We should follow their lead. Meaning if they preserve the original column we should do so here and vice versa. Same for the initial value of `null` etc.

File: docs/SPEC.md [hunk 2]
```diff
@@ -3019,6 +3019,25 @@ from(bucket: "telegraf/autogen")
...
+
+Elapsed has the following properties:
+
+| Name        | Type     | Description                                                                                                                                                 |
+| ----        | ----     | -----------                                                                                                                                                 |
+| unit        | duration | The unit in which the elapsed time is returned. Defaults to `1s`.|
+| timeColumn  | string   | Name of the `flux.TTime` column on which to compute the elapsed time. Defaults to `_time`.                                                                  |
+
+**CURRENTLY**: 
+- no `unit`. default time is nanoseconds.
```
[the remark this answers is not in the document]
Comment (ewendai) [reply docs/SPEC.md]: `derivative` has default time in seconds. so i guess for consistency?

File: docs/SPEC.md [hunk 3]
```diff
@@ -3019,6 +3019,25 @@ from(bucket: "telegraf/autogen")
...
+Elapsed has the following properties:
+
+| Name        | Type     | Description                                                                                                                                                 |
+| ----        | ----     | -----------                                                                                                                                                 |
+| unit        | duration | The unit in which the elapsed time is returned. Defaults to `1s`.|
+| timeColumn  | string   | Name of the `flux.TTime` column on which to compute the elapsed time. Defaults to `_time`.                                                                  |
+
+**CURRENTLY**: 
+- no `unit`. default time is nanoseconds. 
+- Not sure if will have the "returns `0` if `unit` option is
```
[the remark this answers is not in the document]
Comment (nathanielc) [reply docs/SPEC.md]: Should we introduce the concept of a duration column now? Meaning a column that is the same as an int64 in nanoseconds but semantically is a duration column? Much like our time column type? 

Would doing so simplify and clarify how elapsed works?

File: docs/SPEC.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (aanthony1243) [reply docs/SPEC.md]: makes sense.  leave it then.  if we decide on nano, we'll have to make a new issue to change both functions.

File: docs/SPEC.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (aanthony1243) [reply docs/SPEC.md]: ok, then I think that the original column is lost, and the first row is null.  @ewendai can you look at the code/docs for derivative and confirm?

[the remark this answers is not in the document]
Comment (ewendai) [reply docs/SPEC.md]: Derivative computes the time based difference between subsequent non-null records. Difference computes the difference between subsequent records. Looking at `difference` it seems like the first line just disappears? 

Example of difference:

| _time |   A  |   B  |   C  | tag |
|:-----:|:----:|:----:|:----:|:---:|
|  0001 | null |   1  |   2  |  tv |
|  0002 |   6  |   2  | null |  tv |
|  0003 |   4  |   2  |   4  |  tv |
|  0004 |  10  |  10  |   2  |  tv |
|  0005 | null | null |   1  |  tv |

Result (`nonNegative: false`):

| _time |   A  |   B  |   C  | tag |
|:-----:|:----:|:----:|:----:|:---:|
|  0002 | null |   1  | null |  tv |
|  0003 |  -2  |   0  |   2  |  tv |
|  0004 |   6  |   8  |  -2  |  tv |
|  0005 | null | null |  -1  |  tv |

Result (`nonNegative: true`):

| _time |   A  |   B  |   C  | tag |
|:-----:|:----:|:----:|:----:|:---:|
|  0002 | null |   1  | null |  tv |
|  0003 | null |   0  |   2  |  tv |
|  0004 |   6  |   8  | null |  tv |
|  0005 | null | null | null |  tv |

Looking at the end to end test for `difference`, it replaces the contents of `_value` with the differences but keeps the other columns. However, it wouldn't make sense to replace the values of `_time` with the elapsed time, so I feel that we should just add an "elapsed" column?

[the remark this answers is not in the document]
Comment (aanthony1243) [reply docs/SPEC.md]: I think I agree with you.  This function is different from difference and derivative that it doesn't really have to access the _value column to compute its result. I like your proposal:   drop the first row, and add a new `elapsed` column.   Might as well add a `columnName` parameter with default == `elapsed`

File: stdlib/universe/elapsed.go [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+	plan.RegisterProcedureSpec(ElapsedKind, newElapsedProcedure, ElapsedKind)
+	execute.RegisterTransformation(ElapsedKind, createElapsedTransformation)
+}
+
+func createElapsedOpSpec(args flux.Arguments, a *flux.Administration) (flux.OperationSpec, error) {
+	if err := a.AddParentFromArgs(args); err != nil {
+		return nil, err
+	}
+
+	err := a.AddParentFromArgs(args)
```
Comment (jlapacik) [inline stdlib/universe/elapsed.go]: This looks like a duplicate from the line above. I think you can remove it.

File: stdlib/universe/elapsed.go [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+	cols := tbl.Cols()
+	numCol := 0
+
+	err := execute.AddTableCols(tbl, builder)
+	if err != nil {
+		return err
+	}
+
+	ind := -1
+	for i, c := range cols {
```
Comment (jlapacik) [inline stdlib/universe/elapsed.go]: We have a helper function `execute.ColIdx` that I believe does what you're doing here in this for loop. I think you could get replace this loop entirely with that function here. Just to shrink the size of the code.

File: stdlib/universe/elapsed.go (hunk 4 above)
Comment (ewendai) [reply stdlib/universe/elapsed.go]: if i remove line 41, then `err` will never have been declared (b/c it was previously in an `if` block)

**Pull request merged**